cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import time
+import datetime
+try:
+    import boto3
+    import botocore
+except ImportError:
+    module.fail_json(msg='boto3 and botocore are required.')
+from distutils.version import LooseVersion
+if LooseVersion(botocore.__version__) < LooseVersion("1.3.14"):
+    module.fail_json(msg='botocore version 1.3.14 or above is required.')
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_nat_gateway
+short_description: Manage AWS VPC NAT Gateways
+description:
+  - Ensure the state of AWS VPC NAT Gateways based on their id, allocation and subnet ids.
+version_added: "2.1"
+options:
+  state:
+    description:
+      - Ensure NAT Gateway is present or absent
+    required: false
+    default: present
+    choices: ['present', 'absent']
+  nat_gateway_id:
+    description:
+      - The id AWS dynamically allocates to the NAT Gateway on creation
+    required: false
```

Comment:
"If required is false you need to document default, even if the default is ‘None’"